app/data/views.py

```diff
@@ -229,3 +229,37 @@ def get_serializer_class(self):
 
 class DriverBoundaryViewSet(BoundaryViewSet):
     permission_classes = (IsAdminOrReadOnly,)
+
+
+class DriverRecordDuplicateViewSet(viewsets.ModelViewSet):
+    queryset = RecordDuplicate.objects.all().order_by('record__occurred_to')
+    serializer_class = RecordDuplicateSerializer
+    permission_classes = (ReadersReadWritersWrite,)
+    filter_class = filters.RecordDuplicateFilter
+
+    @detail_route(methods=['patch'])
+    def resolve(self, request, pk=None):
+        duplicate = self.queryset.get(pk=pk)
+        resolved = []
+        recordUUID = request.data.get('recordUUID', None)
+        if recordUUID is None:
+            # No record id means they want to keep both, so just resolve the duplicate
+            pass
+        elif recordUUID == str(duplicate.duplicate_record.uuid):
+            # Picking B means all duplicates with primary record A are resolved, not just this one
+            duplicate.record.archived = True
+            duplicate.record.save()
+            for dup in self.queryset.exclude(pk=pk).filter(record=duplicate.record):
```

Comment:
Have you tried this for duplicates that result in a lot of cascading resolutions? I could see this potentially taking a long time in situations where a bunch of records are all duplicates of each other. It might be more efficient to grab the ids of the records with a `.values_list('pk', flat=True)` and then perform a `.update(resolved=True)`, that way it's limited to two queries no matter what. But it may not be a problem, not sure.